src/server/pfs/server/transaction_defer.go

```diff
@@ -56,3 +56,51 @@ func (t *Propagater) Run(ctx context.Context) error {
 	}
 	return t.d.propagateBranches(ctx, t.txnCtx, branches)
 }
+
+type RepoValidator struct {
```

Comment:
I'm concerned about this object being public, and capturing the transaction context.  It seems like it would be easy to misuse.